### PR TITLE
Business Bio Blocks

### PR DESCRIPTION
**Pull request by nateh777**
##### What does the change add or fix?

Bring Person Bio block into repo and convert to specific for businesses. Removing elements that are unnecessary for businesses. Also found I needed to pull in Person Edit and Business Detail blocks. Person Edit got changed into a "Business Edit" block, which again, removed unnecessary fields, like Gender, Marital Status, etc. Matching business detail edit capabilities, but keeping some that they wanted, such as business photos. Added a new setting to Business Detail block to be able to hide the details section, so that the block could be used to just list Business Contacts on its own.

---------

##### What does the change add or fix in a succinct statement that will be read by clients?

- Added new Business Bio Block, removing unnecessary fields for businesses.
- Added new Business Edit Block, removing unnecessary fields for businesses.
- Added a modified Business Detail block with new setting to hide the details section to show the Business Contacts portion of the block on its own..

---------

##### Who reported, requested, or paid for the change?

WCD

---------

##### Does this update or add options to the block UI?

image

image

image

---------

##### What files does it affect?

- Crm/BusinessBio.ascx
- Crm/BusinessBio.ascx.cs
- Crm/EditBusiness.ascx
- Crm/EditBusiness.ascx.cs
- Finance/BusinessDetail.ascx
- Finance/BusinessDetail.ascx.cs

---------

##### Is it a breaking change for other versions/clients?

No

### CODE REVIEW

File: Crm/BusinessBio.ascx.cs [hunk 1]
```diff
@@ -403,60 +392,58 @@ protected override void OnLoad( EventArgs e )
...
                     {
-                        lMaritalStatus.Text = string.Format( "<dd>{0}</dd>", Person.MaritalStatusValueId.DefinedValue() );
+                        var location = business.GivingGroup.GroupLocations
+                            .Where( gl => gl.GroupLocationTypeValueId == workLocationType.Id )
+                            .Select( gl => gl.Location )
+                            .FirstOrDefault();
+                        if ( location != null )
+                        {
+                            var addTo = detailsLeft;
+                            if ( business.GivingGroup != null && business.RecordStatusReasonValue != null )
```
Comment (nateh777) [inline Crm/BusinessBio.ascx.cs:438]: Change to GivingGroup.Campus != null

File: Crm/BusinessBio.ascx.cs [hunk 2]
```diff
@@ -403,60 +392,58 @@ protected override void OnLoad( EventArgs e )
...
+                        //detailsRight.Add( "Phone Number", phoneNumber.ToString() );
+                        detailsRight.Add( "Phone Number", FormatPhoneNumber( phoneNumber.IsUnlisted, phoneNumber.CountryCode, phoneNumber.ToString(), phoneNumber.NumberTypeValueId ?? 0, phoneNumber.IsMessagingEnabled ) );
                     }
-
-                    rptPhones.DataSource = phoneNumbers;
-                    rptPhones.DataBind();
                 }
 
-                var communicationLinkedPageValue = this.GetAttributeValue( AttributeKey.CommunicationPage );
+                var communicationLinkedPageValue = this.GetAttributeValue( "CommunicationPage" );
```
Comment (nateh777) [inline Crm/BusinessBio.ascx.cs:460]: AttributeKey.CommunicationPage

File: Crm/EditBusiness.ascx [hunk 3]
```diff
@@ -20,137 +20,67 @@
...
+                            <Rock:DefinedValuePicker ID="dvpReason" runat="server" Label="Reason" Visible="false"></Rock:DefinedValuePicker>
                             <Rock:ImageEditor ID="imgPhoto" runat="server" Label="Photo" BinaryFileTypeGuid="03BD8476-8A9F-4078-B628-5B538F967AFC" />
                         </div>
                     </div>
 
                     <div class="col-md-9">
 
                         <div class="well form-well">
-                            <fieldset>
+                            <Rock:DataTextBox ID="tbBusinessName" runat="server" Label="Name" SourceTypeName="Rock.Model.Person, Rock" PropertyName="FirstName" />
```
Comment (nateh777) [inline Crm/EditBusiness.ascx:38]: Is PropertyName="FirstName" correct or should it be LastName?

File: Crm/EditBusiness.ascx.cs [hunk 4]
```diff
@@ -359,131 +311,81 @@ protected void btnSave_Click( object sender, EventArgs e )
...
-                        var phoneNumberService = new PhoneNumberService( rockContext );
-                        foreach ( var phoneNumber in person.PhoneNumbers
-                            .Where( n => n.NumberTypeValueId.HasValue && !phoneNumberTypeIds.Contains( n.NumberTypeValueId.Value ) )
-                            .ToList() )
+                        // Record Type - this is always "business". it will never change.
+                        business.RecordTypeValueId = DefinedValueCache.Get( Rock.SystemGuid.DefinedValue.PERSON_RECORD_TYPE_BUSINESS.AsGuid() ).Id;
+
+                        // Record Status
+                        business.RecordStatusValueId = dvpRecordStatus.SelectedValueAsInt();
+                        ;
```
Comment (nateh777) [inline Crm/EditBusiness.ascx.cs]: Does this get cleaned up?

**Review by mattbaylor — APPROVED** [review]

**Pull request merged**